ui/src/components/shared/EnvHoC.tsx

```diff
@@ -0,0 +1,39 @@
+import { FC, PropsWithChildren } from "react";
+
+import { envParser } from "src/adapters/env";
+import { ErrorResult } from "src/components/shared/ErrorResult";
+import { Env } from "src/domain";
+import { processZodError } from "src/utils/adapters";
+
+const envParsed = envParser.safeParse(import.meta.env);
+export const env: Env = envParsed.success
+  ? envParsed.data
+  : {
+      api: {
+        password: "",
+        url: "",
+        username: "",
+      },
+      issuer: {
+        did: "",
+        name: "",
+      },
+    };
+
+export const authorization = `Basic ${env.api.username}:${env.api.password}`;
```

Comment:
Since this is exported, it would be clearer to call this something like `apiAuthHeader`.